src/Bonsai.ML.Torch/OpenCVHelper.cs

```diff
@@ -118,13 +63,21 @@ public static Tensor ToTensor(Mat mat)
             var depth = mat.Depth;
             var tensorType = bitDepthLookup.FirstOrDefault(x => x.Value.Depth == depth).Key;
 
-            IntPtr tensorDataPtr = mat.Data;
-            long[] dimensions = [ height, width, channels ];
-            if (tensorDataPtr == IntPtr.Zero) 
-            {
-                return empty(dimensions);
-            }
-            return CreateTensorFromPtr(tensorDataPtr, dimensions, tensorType);  
+            IntPtr data = mat.Data;
+            ReadOnlySpan<long> dimensions = stackalloc long[] { height, width, channels };
+            if (data == IntPtr.Zero) 
+                return zeros(dimensions);
+
+            return TorchSharpEx.CreateTensorFromUnmanagedMemoryWithManagedAnchor(data, mat, dimensions, tensorType);
+        }
+
+        private static (int height, int width, int channels) GetImageDimensions(this Tensor tensor)
```

Comment:
This method exists to avoid `Tensor.shape` (and by extension `Tensor.size()`) as it is a performance footgun since it creates a bunch of garbage every time you call it.

Consider [banning these APIs](https://github.com/dotnet/roslyn-analyzers/blob/3746107f6c9e0a2f1cae7e37a78a9d9de249857a/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md) in this repo to avoid them. (TorchSharp is not a pit of success when it comes to generating garbage in general though so this might be futile.)